### PR TITLE
Add unique question

### PR DESCRIPTION
**Pull request by yoshiki1009**
implement add and delete unique question feature

### CODE REVIEW

File: client/src/screens/CreateProduct.js [hunk 1]
```diff
@@ -31,7 +33,7 @@ export default (props) => {
   // const [, setImage] = useState(props.route.params.imageUrl)
   const code = image.value.code
   const [highlitedQuestion, setHighlitedQuestion] = useState([])
-  const [uniqQuestion, setUniqQuestion] = useState('')
+  const [uniqQuestion, setUniqQuestion] = useState([''])
```
Comment (satoshinbp) [inline client/src/screens/CreateProduct.js:36]:
```suggestion
  const [uniqQuestions, setUniqQuestions] = useState([])
```

File: client/src/screens/CreateProduct.js [hunk 2]
```diff
@@ -61,9 +63,14 @@ export default (props) => {
...
       uniqQuestion: uniqQuestion,
     }
+
     uploadImage()
     const res = await postProduct(params)
+    props.navigation.navigate('Scan', {})
     props.navigation.navigate('Product', { id: res.data._id })
+    dispatch(deleteProduct())
+    setHighlitedQuestion([])
+    setUniqQuestion('')
```
Comment (satoshinbp) [inline client/src/screens/CreateProduct.js:73]:
```suggestion
    setUniqQuestions([])
```

File: client/src/screens/CreateProduct.js [hunk 3]
```diff
@@ -152,18 +183,25 @@ export default (props) => {
...
-              blurOnSubmit={true}
-              returnKeyType="done"
-              onSubmitEditing={() => {
-                Keyboard.dismiss()
-              }}
-              alignItems="center"
-              value={uniqQuestion}
-              onChangeText={(text) => setUniqQuestion(text)}
-            />
+            {uniqQuestion.map((element, index) => (
```
Comment (satoshinbp) [inline client/src/screens/CreateProduct.js:186]:
```suggestion
            {uniqQuestions.map((uniqQuestion, index) => (
```

File: client/src/screens/CreateProduct.js [hunk 4]
```diff
@@ -152,18 +183,25 @@ export default (props) => {
...
+                <Input
+                  mb="10"
+                  placeholder="Write your own question here"
+                  blurOnSubmit={true}
+                  returnKeyType="done"
+                  onSubmitEditing={() => {
+                    Keyboard.dismiss()
+                  }}
+                  alignItems="center"
+                  value={uniqQuestion[index]}
```
Comment (satoshinbp) [inline client/src/screens/CreateProduct.js:197]:
```suggestion
                  value={uniqQuestion}
```

File: client/src/screens/CreateProduct.js [hunk 5]
```diff
@@ -31,7 +33,7 @@ export default (props) => {
   // const [, setImage] = useState(props.route.params.imageUrl)
   const code = image.value.code
   const [highlitedQuestion, setHighlitedQuestion] = useState([])
```
Comment (satoshinbp) [inline client/src/screens/CreateProduct.js:35]:
```suggestion
  const [highlitedQuestions, setHighlitedQuestions] = useState([])
```

File: client/src/screens/CreateProduct.js [hunk 6]
```diff
@@ -61,9 +63,14 @@ export default (props) => {
       highlitedQuestion: highlitedQuestion,
       uniqQuestion: uniqQuestion,
```
Comment (satoshinbp) [inline client/src/screens/CreateProduct.js:64]:
```suggestion
      highlitedQuestions,
      uniqQuestions,
```

File: client/src/screens/CreateProduct.js [hunk 7]
```diff
@@ -61,9 +63,14 @@ export default (props) => {
...
     }
+
     uploadImage()
     const res = await postProduct(params)
+    props.navigation.navigate('Scan', {})
     props.navigation.navigate('Product', { id: res.data._id })
+    dispatch(deleteProduct())
+    setHighlitedQuestion([])
```
Comment (satoshinbp) [inline client/src/screens/CreateProduct.js:72]:
```suggestion
    setHighlitedQuestions([])
```

File: client/src/screens/CreateProduct.js [hunk 8]
```diff
@@ -91,16 +98,40 @@ export default (props) => {
...
+    } else {
+      props.navigation.navigate('Scan', {})
+    }
+  }
+
+  const handleChange = (i, text) => {
+    const newUniqQuestion = [...uniqQuestion]
+    newUniqQuestion[i] = text
+    setUniqQuestion(newUniqQuestion)
+  }
```
Comment (satoshinbp) [inline client/src/screens/CreateProduct.js:113]:
```suggestion
  const handleChange = (i, text) => {
    const newUniqQuestions = [...uniqQuestions]
    newUniqQuestions[i] = text
    setUniqQuestions(newUniqQuestions)
  }
```

File: client/src/screens/CreateProduct.js [hunk 9]
```diff
@@ -91,16 +98,40 @@ export default (props) => {
...
+
+  const addFormFields = () => {
+    setUniqQuestion([...uniqQuestion, ''])
```
Comment (satoshinbp) [inline client/src/screens/CreateProduct.js:116]:
```suggestion
    setUniqQuestions([...uniqQuestions, ''])
```

File: client/src/screens/CreateProduct.js [hunk 10]
```diff
@@ -91,16 +98,40 @@ export default (props) => {
...
+  }
+
+  const removeFormFields = (i) => {
+    const newUniqQuestion = [...uniqQuestion]
+    newUniqQuestion.splice(i, 1)
+    setUniqQuestion(newUniqQuestion)
+  }
```
Comment (satoshinbp) [inline client/src/screens/CreateProduct.js:123]:
```suggestion
  const removeFormFields = (i) => {
    const newUniqQuestions = [...uniqQuestions]
    newUniqQuestions.splice(i, 1)
    setUniqQuestions(newUniqQuestions)
  }
```

File: controllers/productController.js [hunk 11]
```diff
@@ -53,14 +53,12 @@ const createProduct = (req, res) => {
...
 
-    uniqQandAs: [
-      {
-        question: {
-          description: req.body.uniqQuestion,
-        },
-        highlightedBy: req.body.uniqQuestion ? [req.body.userId] : [],
+    uniqQandAs: req.body.uniqQuestion.map((uniqquestion, index) => ({
+      question: {
+        description: uniqquestion,
```
Comment (satoshinbp) [inline controllers/productController.js:58]:
```suggestion
    uniqQandAs: req.body.uniqQuestions.map((uniqQuestion, index) => ({
      question: {
        description: uniqQuestion,
```

**Review by satoshinbp — CHANGES_REQUESTED** [review]
Use plural case for variable: array for better readability. 
Please resolve conflicts also.

**Comment by yoshiki1009** [comment]
fixed variable naming and confliction

**Review by satoshinbp — APPROVED** [review]

**Pull request merged**